libraries/AP_Proximity/AP_Proximity_MR72_CAN.cpp

```diff
@@ -34,10 +34,13 @@ const AP_Param::GroupInfo AP_Proximity_MR72_CAN::var_info[] = {
 
 AP_Proximity_MR72_CAN::AP_Proximity_MR72_CAN(AP_Proximity &_frontend,
                                      AP_Proximity::Proximity_State &_state,
-                                     AP_Proximity_Params& _params):
+                                     AP_Proximity_Params& _params,
+                                     Manufacturer manufacture):
     AP_Proximity_Backend(_frontend, _state, _params)
 {
-    multican_MR72 = NEW_NOTHROW MultiCAN{FUNCTOR_BIND_MEMBER(&AP_Proximity_MR72_CAN::handle_frame, bool, AP_HAL::CANFrame &), AP_CAN::Protocol::NanoRadar, "MR72 MultiCAN"};
+    // determine CAN protocol based on manufacturer
+    AP_CAN::Protocol can_protocol = manufacture == Manufacturer::NanoRadar ? AP_CAN::Protocol::NanoRadar : AP_CAN::Protocol::Hexsoon;
```

Comment:
```suggestion
    AP_CAN::Protocol can_protocol = ((manufacture == Manufacturer::NanoRadar) ? AP_CAN::Protocol::NanoRadar : AP_CAN::Protocol::Hexsoon);
```

Reply:
I'm finding it *really* strange that we would call a CAN protocol "Hexsoon".  I mean... that's a very limiting thing to do!

I'd prefer if we had AP_CAN::Protocol::NanoRadarMR72 and AP_CAN::Protocol::HexsoonMR72 (or at least something else to identify it.

Reply:
We also call the sensors from NanoRadars CAN protocol, "NanoRadar" and also "Benewake" and so on...

Reply:
I've renamed the existing "NanoRadar" to "RadarCAN" so no need to worry about the "Hexsoon" can protocol any more